### ISSUE TITLE
Unexpected warning for explicit 'in' argument

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#47000]
**Version Used**: ca88e94973680f1614777a22afe136d345e7767e

**Steps to Reproduce**:

```csharp
#nullable enable

public class C {
    public void M() {
        object? a = null;
        object b = new object();
        
        Target(a, b);
        Target(in a, b);
        Target(a, in b);
        Target(in a, in b); // <-- warning here
    }
    
    public void Target<T>(in T v1, in T v2) { }
}
```

**Expected Behavior**:

Compile without warnings.

**Actual Behavior**:

The following warning is reported _only_ for the call where both `a` and `b` are passed with explicit `in`.

> warning CS8604: Possible null reference argument for parameter 'v1' in 'void C.Target&lt;object>(in object v1, in object v2)'.
